From c907e83a58f79c2963152a0dba90d495a1506cdb Mon Sep 17 00:00:00 2001
From: Scott Jaderholm <jaderholm@gmail.com>
Date: Sat, 25 Mar 2017 16:34:05 -0700
Subject: [PATCH] favicon: Add filler callback and principal

Somewhere around version 50, Firefox started giving a deprecation
warning if these args, marked optional or missing in MDN documentation,
weren't present.
---
 modules/favicon.js | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/modules/favicon.js b/modules/favicon.js
index d1f0d92..aae69cd 100644
--- modules/favicon.js
+++ modules/favicon.js
@@ -23,7 +23,8 @@ define_variable("favicon_image_max_size", 1024,
         favicon_set_internal = function (buffer, icon_url) {
             favicon_service.setAndFetchFaviconForPage(
                 buffer.current_uri, icon_url, false,
-                favicon_service.FAVICON_LOAD_NON_PRIVATE);
+                favicon_service.FAVICON_LOAD_NON_PRIVATE,
+                function() { }, buffer.document.nodePrincipal);
         };
     } else {
         favicon_set_internal = function (buffer, icon_url) {
-- 
2.7.4.GIT

